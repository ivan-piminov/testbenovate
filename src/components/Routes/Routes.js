import React from 'react';
import {Redirect, Route} from "react-router-dom";
import {StartPage} from "../StartPage/StartPage";
import {CommonTable} from "../CommonTable/CommonTable";
import {ChoiceButtons} from "../ChoiceButtons/ChoiceButtons";
import {SortTable} from "../SortTable/SortTable";


export const MAIN = '/main'
export const TABLES = '/tables'
export const COMMON_TABLE = '/common_table'
export const SORT_TABLE = '/sort_table'


export const Routes = (props) => {
    return (
        <div>
            <Route exact path='/' render={() => <Redirect to={MAIN}/>}/>
            <Route path={MAIN} render={() => <StartPage isLoading={props.isLoading}/>}/>
            <Route path={TABLES} render={() => <ChoiceButtons/>}/>
            <Route path={COMMON_TABLE} render={() => <CommonTable isLoading={props.isLoading} data = {props.data} onSort = {props.onSort} sort = {props.sort} field = {props.sortField} addNewWorker={props.addNewWorker} />}/>
            <Route path={SORT_TABLE} render={() => <SortTable isLoading={props.isLoading}  data = {props.data}/>}/>
        </div>
    )
}
