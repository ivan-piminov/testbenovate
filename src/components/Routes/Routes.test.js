import React from "react";
import {Routes} from './Routes';
import {shallow} from "enzyme";

describe('Testing <Routes/>',()=>{
    it('Routes render correctly', ()=>{
        const routes = shallow(<Routes/>);
        expect(routes).toMatchSnapshot();
    })
});