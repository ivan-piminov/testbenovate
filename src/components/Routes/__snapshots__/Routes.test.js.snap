// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Routes/> Routes render correctly 1`] = `ShallowWrapper {}`;
